source/main/utils/handleCheckBlockReplayProgress.js

```diff
@@ -20,19 +55,26 @@ export const handleCheckBlockReplayProgress = (
     const rl = readline.createInterface({ input: fileStream });
     const progress = [];
     for await (const line of rl) {
-      if (line.includes('block replay')) {
+      if (containProgressKeywords(line)) {
         progress.push(line);
       }
     }
     if (!progress.length) return;
 
     const finalProgress = progress.slice(-1).pop();
```

Comment:
This can be simplified (removed) by remembering just a single string representing the last progress line instead of an array of all such lines.